pf/tfbridge/ignore_changes.go

```diff
@@ -265,35 +265,30 @@ func (*matchingContext) fail() *matchingContext {
 	}
 }
 
-func (*matchingContext) pulumiNameMatches(pattern interface{}, puName resource.PropertyKey) bool {
-	if p, ok := pattern.(string); ok {
-		if p == "*" {
-			return true
-		}
-		if p == string(puName) {
-			return true
-		}
-	}
-	return false
+func (mc *matchingContext) pulumiNameMatches(pattern interface{}, puName resource.PropertyKey) bool {
```

Comment:
Maybe drop ignore_changes from this PR and instead let's go for fixing https://github.com/pulumi/pulumi-terraform-bridge/issues/920 using Pulumi-level approach you outlined, that's going to be oblivious to sets anyway.